src/fKeyTexts.pas

```diff
@@ -211,6 +165,77 @@ procedure TfrmKeyTexts.btnOKClick(Sender: TObject);
   cqrini.SaveToDisk;
   ModalResult := mrOK
 end;
+procedure TfrmKeyTexts.LoadMsgClick(Sender: TObject);
+var
+  section : String = '';
+  CwM  : TMemIniFile;
+begin
+ OpenDialog1.InitialDir:= dmData.HomeDir;
+ if OpenDialog1.Execute then
+  begin
+  CWM := TMemIniFile.Create(OpenDialog1.Filename);
+  section := 'CW';
+  edtF1.text     := CWM.ReadString(section,'F1','cq cq de %mc %mc pse K');
+  edtF2.text     := CWM.ReadString(section,'F2','');
+  edtF3.text     := CWM.ReadString(section,'F3','');
+  edtF4.text     := CWM.ReadString(section,'F4','');
+  edtF5.text     := CWM.ReadString(section,'F5','');
+  edtF6.text     := CWM.ReadString(section,'F6','');
+  edtF7.text     := CWM.ReadString(section,'F7','');
+  edtF8.text     := CWM.ReadString(section,'F8','');
+  edtF9.text     := CWM.ReadString(section,'F9','');
+  edtF10.text    := CWM.ReadString(section,'F10','');
+  edtCapF1.text  := CWM.ReadString(section,'CapF1','F1 - CQ');
+  edtCapF2.text  := CWM.ReadString(section,'CapF2','F2');
+  edtCapF3.text  := CWM.ReadString(section,'CapF3','F3');
+  edtCapF4.text  := CWM.ReadString(section,'CapF4','F4');
+  edtCapF5.text  := CWM.ReadString(section,'CapF5','F5');
+  edtCapF6.text  := CWM.ReadString(section,'CapF6','F6');
+  edtCapF7.text  := CWM.ReadString(section,'CapF7','F7');
+  edtCapF8.text  := CWM.ReadString(section,'CapF8','F8');
+  edtCapF9.text  := CWM.ReadString(section,'CapF9','F9');
+  edtCapF10.text := CWM.ReadString(section,'CapF10','F10');
+  rgEnter.ItemIndex := CWM.ReadInteger(section,'EnterFunction',1);
+  FreeAndNil(CWM);
+  end;
+end;
+
+procedure TfrmKeyTexts.SaveMsgClick(Sender: TObject);
+var
+  section : String = '';
+  CwM  : TMemIniFile;
+ begin
+  SaveDialog1.InitialDir:= dmData.HomeDir;
+  if SaveDialog1.Execute then
+     begin
+      CWM := TMemIniFile.Create(SaveDialog1.Filename);
```

Comment:
Could you put it to try finally block, please?

`
CWM := TMemIniFile.Create(SaveDialog1.Filename);
try
  ....
finally
  FreeAndNil(CWM);
end;

The same in reading. Thanks a lot.
`

Reply:
Good Idea!
But leads to problem.
if doing as you say we do not have SaveDialog1.Filename yet. 
if TMemIniFile.Create is inside try section and section fails we are going to have problem when finally gets executed and there is freeAndNill of object that may not have created at all (if try fails at  TMemIniFile.Creation).

Need someting like  "if CWM<>nil " then FreeAndNil(CWM)" for finally part but not found yet.